axelor-base/src/main/resources/views/Language.xml

```diff
@@ -6,14 +6,18 @@
   <grid name="language-grid" title="Languages" model="com.axelor.apps.base.db.Language"
     editable="true">
     <field name="name"/>
-    <field name="code"/>
+    <field name="code" onChange="action-language-method-convert-to-lowercase"/>
   </grid>
 
   <form name="language-form" title="Language" model="com.axelor.apps.base.db.Language">
     <panel name="mainPanel">
       <field name="name"/>
-      <field name="code"/>
+      <field name="code" onChange="action-language-method-convert-to-lowercase"/>
```

Comment:
this action can be replaced by a `x-bind` attribute, I think `x-bind="{{code|lowercase}}"` should work.